sherpa/ui/utils.py

```diff
@@ -319,103 +319,93 @@ def clean(self):
+            'fit': [sherpa.plot.FitPlot()],
+            'resid': [sherpa.plot.ResidPlot()],
+            'ratio': [sherpa.plot.RatioPlot()],
+            'delchi': [sherpa.plot.DelchiPlot()],
+            'chisqr': [sherpa.plot.ChisqrPlot()],
+            'psf': [sherpa.plot.PSFPlot()],
+            'kernel': [sherpa.plot.PSFKernelPlot()]
         }
 
-        self._plot_type_names = {
-            'data': 'data',
-            'model': 'model',
-            'source': 'source',
-            'fit': 'fit',
-            'resid': 'resid',
-            'ratio': 'ratio',
-            'delchi': 'delchi',
-            'chisqr': 'chisqr',
-            'psf': 'psf',
-            'kernel': 'kernel',
-            'source_component': 'source_component',
-            'model_component': 'model_component',
-            'compsource': 'source_component',
-            'compmodel': 'model_component',
+        # Set up aliases so that calls to set_xlog/.. will still
+        # succeed, but the user will be told to use the new
+        # name. These keys are also used, along with _plot_type_names,
+        # to determine the set of forbidden identifiers.
+        #
+        self._plot_types_alias = {
+            "compsource": "source_component",
+            "compmodel": "model_component"
         }
 
+        # This is used by the get_<key>_contour calls to access the
+        # relevant contour class. The keys define the labels that can be
+        # used in calls to contour(), and are also used to determine
+        # the set of forbidden identifiers.
+        #
         self._contour_types = {
-            'data': self._datacontour,
-            'model': self._modelcontour,
-            'source': self._sourcecontour,
-            'fit': self._fitcontour,
-            'resid': self._residcontour,
-            'ratio': self._ratiocontour,
-            'psf': self._psfcontour,
-            'kernel': self._kernelcontour
-        }
-
-        self._contour_type_names = {
-            'data': 'data',
-            'model': 'model',
-            'source': 'source',
-            'fit': 'fit',
-            'resid': 'resid',
-            'ratio': 'ratio',
-            'psf': 'psf',
-            'kernel': 'kernel',
+            "data": sherpa.plot.DataContour(),
+            "model": sherpa.plot.ModelContour(),
+            "source": sherpa.plot.SourceContour(),
+            "fit": sherpa.plot.FitContour(),
+            "resid": sherpa.plot.ResidContour(),
+            "ratio": sherpa.plot.RatioContour(),
+            "psf": sherpa.plot.PSFContour(),
+            "kernel": sherpa.plot.PSFKernelContour()
         }
 
+        # This is used by the get_<key>_image calls to access the
+        # relevant image class. The keys are not included in any check
+        # of valid identifiers. unlike the plot and contour cases, as
+        # there is
+        #
+        # - no image() call that acts like plot() or contour();
```

Comment:
but there will be - so maybe we should include it already to make sure nobody writes code that uses kays we won't allow later?

Reply:
At the moment the keys for contour and image are already included in the plot table, so it turns out that adding the contour keys into the "what is a valid identifier" doesn't currently add anything, but is done partly for documentation and just-in-case something changes.